### PR TITLE
Bump MSRV to 1.64

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

Bump MSRV to 1.64

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,15 @@ members = [
...
 resolver = "2"
 
+[workspace.package]
+version = "0.8.1"
+authors = ["thin-edge.io team <info@thin-edge.io>"]
+edition = "2021"
+rust-version = "1.64"
+license = "Apache-2.0"
+homepage = "https://thin-edge.io"
+repository = "https://github.com/thin-edge/thin-edge.io"
```
Comment (albinsuresh) [inline Cargo.toml:20]: We define all this common metadata here and inherit them from here in individual crate's Cargo.toml

File: .github/workflows/build-workflow.yml [hunk 2]
```diff
@@ -71,7 +71,7 @@ jobs:
           fetch-depth: 0
 
       - name: Enable toolchain via github action
-        uses: dtolnay/rust-toolchain@1.63.0
+        uses: dtolnay/rust-toolchain@1.64.0
```
Comment (albinsuresh) [inline .github/workflows/build-workflow.yml:74]: I wish I could define the version numbers of all these toolchains spread across all these workflows at a single place. If anyone knows some trick, let me know.

File: crates/common/batcher/src/batcher.rs [hunk 3]
```diff
@@ -108,13 +108,9 @@ impl<B: Batchable> Batcher<B> {
...
-        for batch in &mut self.batches {
-            if batch.batch_start() <= event_time && event_time <= batch.batch_end() {
-                return Some(batch);
-            }
-        }
-
-        None
+        self.batches
+            .iter_mut()
+            .find(|batch| batch.batch_start() <= event_time && event_time <= batch.batch_end())
```
Comment (didier-wenzek) [inline crates/common/batcher/src/batcher.rs:113]: Okay. However, one is not better than the other. And this is why I have a mix feeling to treat clippy warnings as errors.

**Review by didier-wenzek — APPROVED** [review]
Approved

File: .github/workflows/build-workflow.yml (hunk 2 above)
Comment (rina23q) [reply .github/workflows/build-workflow.yml:74]: We can try this feature together. https://docs.github.com/en/actions/learn-github-actions/variables#defining-configuration-variables-for-multiple-workflows

**Review by rina23q — APPROVED** [review]
Thank you!!!

**Pull request merged**